rclcpp/src/rclcpp/node_interfaces/node_graph.cpp

```diff
@@ -134,6 +134,39 @@ NodeGraph::get_service_names_and_types() const
   return services_and_types;
 }
 
+std::map<std::string, std::vector<std::string>>
+NodeGraph::get_service_names_and_types_by_node(
+  const std::string & node_name,
+  const std::string & namespace_) const
+{
+  rcl_names_and_types_t service_names_and_types {};
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  rcl_ret_t ret = rcl_get_service_names_and_types_by_node(
```

Comment:
Check the return code for an error.

Reply:
Absolutely - done: 1b39f50a788ebdc230dab11376b78b942e9f9b08.